backport: Replace fast sync with batch processing

gitferry opened the pull request:
Notable changes:
* remove fast sync completely
* introduce batch process for catching up
* introduce signature submission interval to config
* introduce batch submission size to config

SebastianElvis reviewed (approved):
Thanks! CI failed though

gitferry commented:
@bap2pecs Do you have any ideas why op_e2e_test failed?

bap2pecs commented:
> @bap2pecs Do you have any ideas why op_e2e_test failed?

i think it's due to some of our test cases depend on the fast sync logic. it probably need some non-trivial refactoring. maybe we can just disable the op_e2e_test in CI for now?

bap2pecs commented on clientcontroller/babylon/babylon.go:
```diff
@@ -145,7 +146,125 @@ func (bc *BabylonController) RegisterFinalityProvider(
+}
+
+// UnjailFinalityProvider sends an unjail transaction to the consumer chain
+func (bc *BabylonController) UnjailFinalityProvider(fpPk *btcec.PublicKey) (*types.TxResponse, error) {
```
why is this added in this PR?

gitferry replied:
Opps, should remove it

gitferry commented:
> i think it's due to some of our test cases depend on the fast sync logic. it probably need some non-trivial refactoring. maybe we can just disable the op_e2e_test in CI for now?

Good. Removed it and added todos in ci

bap2pecs commented on finality-provider/config/config.go:
```diff
@@ -55,21 +54,21 @@ var (
-	SyncFpStatusInterval     time.Duration `long:"syncfpstatusinterval" description:"The duration of time that it should sync FP status with the client blockchain"`
+	LogLevel string `long:"loglevel" description:"Logging level for all subsystems" choice:"trace" choice:"debug" choice:"info" choice:"warn" choice:"error" choice:"fatal"`
+	// ChainType and ChainID (if any) of the chain config identify a consumer chain
+	ChainType                   string        `long:"chaintype" description:"the type of the consumer chain" choice:"babylon"`
```
ah it's caused this line change

``
Whoops. There was an error while executing your fpd CLI 'failed to load configuration: /home/finality-provider/.fpd/fpd.conf:6: Invalid value `OPStackL2' for option `--chaintype'. Allowed values are: 'bash-5.1$ fpd commit-pubrand -h
```

bap2pecs commented on finality-provider/service/fp_instance.go:
```diff
@@ -530,55 +393,52 @@ func (fp *FinalityProviderInstance) reportCriticalErr(err error) {
-			res, err = fp.SubmitFinalitySignature(targetBlocks[0])
-		} else {
+		select {
+		case <-time.After(fp.cfg.SubmissionRetryInterval):
```
ok I just found another problem. this will wait `SubmissionRetryInterval` even at the first iteration. Default value is set to 300 seconds. that's why I found out batch sig submission is very slow

I will send out a fix soon

bap2pecs replied:
https://github.com/babylonlabs-io/finality-provider/pull/217
